RFCs/RFC-001: Introspect streams.md

```diff
@@ -0,0 +1,62 @@
+# RFC-001: Introspect Streams
+
+**To be Reviewed By:** 11/19/2019
+
+**Authors:** Swapnil Bawaskar
+
+**Status:** **Active** | Accepted | Dropped | Superseded
+
+**Superseded by:** N/A
+
+**Related:** N/A
+
+
+## Problem
+When the streaming runtime is installed, riff users can create streams (backed by a stream provider e.g. kafka). Users can write functions that read from and write to these streams,
+but there is no easy mechanism to get events into the stream and look into the contents of the stream which are helpful while developing/demoing.
+
+### Anti-Goals
+We will only address this problem for development/demos, not production, so topics like auth/authz are out of scope for this document.
+
+## Solution
+### Ideal solution:
+The  is to provide users with only one cli (riff cli) that will enable users to send/receive messages to/from a stream (commands below). We will have to ask the users to port-forward the gateway service if they are running outside the cluster.
+
+#### User Impact
+We should introduce the following commands to the riff cli:
+- send a massage to the stream:
+
+    `riff streaming stream publish <stream-name> --payload <payload-as-string> --headers <headers-as-string>`  
+    content type has already been specified while creating the stream.
```

Comment:
The concrete concrete content-type of a particular piece of payload may be different (a compatible subtype of) from the stream content-type.

Reply:
This is a good example of why content-type is a misleading field name for Streams.

Reply:
Should the consumer care about the subtype? If not, may be we should just publish using the content-type on the stream. 
Let's start off without the sub-type being specified on this command for now. We can add it later if desired.

Reply:
each message has its own content-type.

> may be we should just publish using the content-type on the stream

This furthers my point that the name content-type on the stream is confusing. It's not a single MIME type. It's more analogous to the HTTP Accept header.

Reply:
The stream doesn't necessarily have a *concrete* content-type (although for the time being this is all we support because we implement "compatibility" as simple "equality").
A stream may have one or several mime-types, including wildcard ones, so you can't always use that as the content-type of the actual, concrete piece of data you publish.